### PR TITLE
13045 13202 13207 13203 3204 13206, Setup Draft SOW route to show page

### PR DESCRIPTION
**Pull request by CeciliaCY**
### Summary
13045 Setup DSOW Route, edit components and list on project(s) page
13203 Setup save for Draft SOW package
13207 Add Attachments section to Draft SOW
13202 Display Package Notes in Draft SOW
13204 Setup submission for Draft SOW
13206 Setup show page for Draft SOW

#### Task/Bug Number
Close [AB#13045](https://dcp-paperless.visualstudio.com/ZAP%20Portals/_workitems/edit/13045)
Close [AB#13203](https://dcp-paperless.visualstudio.com/ZAP%20Portals/_workitems/edit/13203)
Close [AB#13207](https://dcp-paperless.visualstudio.com/ZAP%20Portals/_workitems/edit/13207)
Close [AB#13202](https://dcp-paperless.visualstudio.com/ZAP%20Portals/_workitems/edit/13202)
Close [AB#13204](https://dcp-paperless.visualstudio.com/ZAP%20Portals/_workitems/edit/13204)
Close [AB#13206](https://dcp-paperless.visualstudio.com/ZAP%20Portals/_workitems/edit/13206)

### Technical Explanation
As all these tasks are related to Draft SOW, instead of implement it one by one, I combined all into one branch and one pull request. Related tech can be found:
#794 
#802 
#805 
#818 
#832 
#825

### CODE REVIEW

File: client/app/components/packages/scope-of-work-draft/edit.hbs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+<SaveableForm
```
Comment (godfreyyeung) [inline client/app/components/packages/scope-of-work-draft/edit.hbs:1]: why is `draft` at the end of the folder title here?
Why not follow the previous pattern of `draft-scope-of-work/edit.hbs`?
Like `draft-eas/edit.hbs`?

File: client/app/components/packages/scope-of-work-draft/edit.hbs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  @validators={{array (hash) (hash)}} 
+  as |saveableForm|
+>
+
+  <div class="grid-x grid-margin-x">
+    <div class="cell large-8">
+
+      <section class="form-section">
+        <h1 class="header-large">
+          Draft Scope of Work (SOW)
```
Comment (godfreyyeung) [inline client/app/components/packages/scope-of-work-draft/edit.hbs:12]: Should be `(DSOW)`,  not `(SOW)`, right?

File: client/app/components/packages/scope-of-work-draft/edit.hbs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          isEnabled=saveableForm.isSubmittable
+          class="no-margin"
+        }}
+        @footer={{component 'packages/scope-of-work-draft/scope-of-work-draft-error'
+          package=@package
+        }}
+        @continueButtonTitle="Continue Editing"
+        data-test-confirm-submit-button={{true}}
+      >
+        <h6>Confirm Draft Scope of Work (SOW) Submission</h6>
```
Comment (godfreyyeung) [inline client/app/components/packages/scope-of-work-draft/edit.hbs:81]: (DSOW) not (SOW)? If you agree with (DSOW), can you update it in all other places, too?

File: client/app/components/packages/scope-of-work-draft/edit.hbs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        />
+      </saveableForm.PageNav>
+
+      <saveableForm.ConfirmationModal
+        @action={{component saveableForm.SubmitButton
+          onClick=this.submitPackage
+          isEnabled=saveableForm.isSubmittable
+          class="no-margin"
+        }}
+        @footer={{component 'packages/scope-of-work-draft/scope-of-work-draft-error'
```
Comment (godfreyyeung) [inline client/app/components/packages/scope-of-work-draft/edit.hbs:75]: The folder naming needs to be consistent. Please change this to `draft-scope-of-work/draft-scope-of-work-error`

File: client/app/components/packages/scope-of-work-draft/edit.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default class PackagesScopeOfWorkDraftEditComponent extends Component {
```
Comment (godfreyyeung) [inline client/app/components/packages/scope-of-work-draft/edit.js:5]: The folder naming needs to be consistent. Please change this file to `draft-scope-of-work/edit.js` and the Component to PackagesDraftScopeOfWorkEditComponent

**Review by godfreyyeung — CHANGES_REQUESTED** [review]
Thank you for all this work. However the naming pattern is inconsistent with the rest of the app, and also inconsistent within this PR (sometimes you use Scope of Work Draft, sometimes Draft Scope of Work). 

Can you change all references to Draft Scope of Work? 
Or is there a particular reason for writing Scope of Work Draft?

File: client/app/components/packages/scope-of-work-draft/edit.hbs (hunk 1 above)
Comment (CeciliaCY) [reply client/app/components/packages/scope-of-work-draft/edit.hbs:1]: @godfreyyeung , this is suggested by @allthesignals

File: client/app/components/packages/scope-of-work-draft/edit.hbs (hunk 2 above)
Comment (CeciliaCY) [reply client/app/components/packages/scope-of-work-draft/edit.hbs:12]: Wireframe has (SOW), not (DSOW)

File: client/app/components/packages/scope-of-work-draft/edit.hbs (hunk 4 above)
Comment (CeciliaCY) [reply client/app/components/packages/scope-of-work-draft/edit.hbs:75]: Let's check with @allthesignals first!

File: client/app/components/packages/scope-of-work-draft/edit.js (hunk 5 above)
Comment (CeciliaCY) [reply client/app/components/packages/scope-of-work-draft/edit.js:5]: @godfreyyeung Let's check with @allthesignals first:)

**Comment by CeciliaCY** [comment]
> Thank you for all this work. However the naming pattern is inconsistent with the rest of the app, and also inconsistent within this PR (sometimes you use Scope of Work Draft, sometimes Draft Scope of Work).
> 
> Can you change all references to Draft Scope of Work? And also change the acronyms to all be DSOW
> Or is there a particular reason for writing Scope of Work Draft?

@godfreyyeung when it's in the code (such as component name, function name), I use Scope of Work Draft. When it's the plain text which displayed to users, then I use Draft Scope of Work. Let me know the final decision regarding the naming pattern after you have a discussion with @allthesignals

File: client/app/components/packages/scope-of-work-draft/edit.js (hunk 5 above)
Comment (godfreyyeung) [reply client/app/components/packages/scope-of-work-draft/edit.js:5]: Okay, Just had a conversation with @allthesignals . You're good to go!
We're planning on doing some extra reorganizing later to use folders to dilineate draft vs final, etc.
e.g. `sow/draft.hbs` and `sow/final.hbs`, etc.

**Comment by godfreyyeung** [comment]
> > Thank you for all this work. However the naming pattern is inconsistent with the rest of the app, and also inconsistent within this PR (sometimes you use Scope of Work Draft, sometimes Draft Scope of Work).
> > Can you change all references to Draft Scope of Work? And also change the acronyms to all be DSOW
> > Or is there a particular reason for writing Scope of Work Draft?
> 
> @godfreyyeung when it's in the code (such as component name, function name), I use Scope of Work Draft. When it's the plain text which displayed to users, then I use Draft Scope of Work. Let me know the final decision regarding the naming pattern after you have a discussion with @allthesignals

Thank you for the explanation @CeciliaCY ! Sorry I missed that conversation earlier!

**Review by godfreyyeung — APPROVED** [review]

File: client/app/components/project/package-list-item.hbs [hunk 6]
```diff
@@ -149,7 +149,35 @@
...
+      {{else}}
+        <strong>
+          Draft SOW
+          <small>
+            Version {{@package.dcpPackageversion}}
+          </small>
+        </strong>
+      {{/if}}
+    </LinkTo>
+  {{/if}}
```
Comment (allthesignals) [inline client/app/components/project/package-list-item.hbs:180]: @godfreyyeung this component is where we have a lot of opportunity for cleanup and de-duplication.

File: client/app/components/packages/scope-of-work-draft/edit.hbs (hunk 4 above)
Comment (allthesignals) [reply client/app/components/packages/scope-of-work-draft/edit.hbs:75]: @godfreyyeung yep, I know we had a conversation, but @CeciliaCY asked about this in chat and I suggested scope-of-work-draft. I still stand by that, as you know! TYVM all!

File: client/app/components/packages/scope-of-work-draft/edit.hbs (hunk 3 above)
Comment (CeciliaCY) [reply client/app/components/packages/scope-of-work-draft/edit.hbs:81]: I missed this message. And just confirmed with Aline, we should use DSOW. Could you please create a task for this change. I will work on this next Monday! Thanks! @godfreyyeung

**Pull request merged**